src/Functions/extractAllGroups.h

```diff
@@ -82,7 +82,8 @@ class FunctionExtractAllGroups : public IFunction
             throw Exception("Length of 'needle' argument must be greater than 0.", ErrorCodes::BAD_ARGUMENTS);
 
         using StringPiece = typename Regexps::Regexp::StringPieceType;
-        const auto & regexp = Regexps::get<false, false>(needle)->getRE2();
+        auto holder = Regexps::get<false, false>(needle);
```

Comment:
Here was a bug. The ownership of an object from pool was not hold.

Reply:
Thanks for fixing, could you please also add a comment to `Regexps::get` that caller has to store the result by value?

Also, I don't quite understand why const reference doesn't suffice here, as it supposed to extend lifetime of temporary object (the unnamed return value) as described in [The most important const](https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/l).

Reply:
We have to hold
```
Regexps::get<false, false>(needle)
```

but we are using
```
Regexps::get<false, false>(needle)->getRE2()
```
instead.

Reply:
It returns `Pool::Pointer` so the reader has to ask: "what is pool?", "why do we have pool here?", "how do I take and return back objects into pool?".

I will add a comment.

Reply:
Added a comment in master.

Reply:
oh, I get it now, thanks